docs/3.0rc/resources/upgrade-prefect-3.mdx

```diff
@@ -0,0 +1,233 @@
+    # The flow will not exit until 
+    # these futures are resolved
+    futures = my_task.map([1, 2, 3, 4])
+```
+
+```python Prefect 3
+@task
+def my_task():
+    ...
+
+
+@flow
+def my_flow():
+    futures = my_task.map([1, 2, 3, 4])
+    # You must wait for futures before exiting flow.
+    futures.wait()
+```
+</CodeGroup>
+
+If you do not wait for the futures, the tasks may fail with a 'Flow must be running to start a task' error. 
+The one exception to this case is that `PrefectFuture`s returned from a flow will resolve to states before completing the flow.
+
+## Task caching and result persistence
+
+Prefect 3 supports transactional semantics - tasks runs can be grouped into larger transactions, enabling more resilient workflows.
+In keeping with the transactional model, Prefect 3 encourages task runs to be idempotent - intended to run only once per flow run.
+
+To support this, Prefect 3 has more eager, intelligent task caching defaults than Prefect 2.
+Whenever a task runs, Prefect 3 first computes its cache key, then uses that cache key to evaluate whether that same task has run with the same inputs and as part of the same flow before. 
+If it has, the cached result from the previous run is used.
+If it hasn't, the task runs. 
+This means that tasks that may have re-run with the same inputs in Prefect 2 will not re-run in Prefect 3 unless the cache key is modified to be more precise.
+The cache key can be configured through the new global `cache_policy` setting, or through the `cache_policy` on a particular task.
+If a particular task cannot be made idempotent, you can disable this behavior by setting `cache_policy=NONE` on the task.
+
+Results and cache keys are not stored with your Prefect Cloud account or Prefect server instance. 
+They are persisted entirely in your configured result location. 
+
+This caching behavior will only take effect when result persistence is enabled.
+The simplest way to enable result persistence is through the Prefect setting `PREFECT_RESULTS_PERSIST_BY_DEFAULT=true`.
+See the [caching documentation](/3.0rc/develop/write-tasks#caching) for more information.
+
+## Notifications are now automations
+
+The notifications feature in Prefect 2 has been expanded to enable many kinds of “if this, then that” logic that can be defined and executed by a Prefect server instance or Prefect Cloud.
+See the [Automations documentation](/3.0rc/automate/events/automations-triggers) for more information.
+
+## Integrations must be upgraded
+
+Integrations compatible with Prefect 2 generally cannot be used with Prefect 3 and vice versa. 
+You must update all of your integrations packages in conjunction with upgrading from Prefect 2 to Prefect 3. 
+We recommend specifying extras when you install the `prefect` package to ensure that you install the integration package version that corresponds to your Prefect version. For example, use `pip install prefect[aws] --pre` to install Prefect 3 and a compatible prefect-aws version.
```

Comment:
I feel like this should be nearer the top alongside any other installation instructions

Reply:
Done!